### PR TITLE
Support --tasks-json for v3

### PR DESCRIPTION
**Pull request by sttk**
By this PR, gulp v3 also support --tasks-json option.

### CODE REVIEW

**Comment by sttk** [comment]
I've modified npm version to 4.0.1 to avoid EPERM error. npm 4.0.1 is the latest version which supports node 0.10. (See [the release note of npm 4.0.2](https://github.com/npm/npm/releases/tag/v4.0.2)).

In addition, I've added `skip_branch_with_pr` key in appveyor.yml not to build feature branch with open PR.

**Comment by phated** [comment]
@sttk Thanks for sending this as a separate PR.  I'm don't have a lot of time to review this currently but I'll keep it at the top of my queue.

File: lib/versioned/^3.7.0/taskTree.js [hunk 1]
```diff
@@ -4,14 +4,22 @@ module.exports = function(tasks) {
   var map = {};
   var arr = [];
   Object.keys(tasks).forEach(function(taskname) {
-    var task = { label: taskname, nodes: [], };
+    var task = {
+      label: taskname,
+      type: 'task',
```
Comment (phated) [inline lib/versioned/^3.7.0/taskTree.js]: @sttk why did you add `type` to this structure?

Comment (sttk) [reply lib/versioned/^3.7.0/taskTree.js]: For getting same formatted json data by `--tasks-json` with gulp v4.

An example of json data is as follows:
```json
{"label":"Tasks for ~/project/gulpfile.js","nodes":[
    {"label":"taskA","type":"task","nodes":[
        {"label":"<parallel>","type":"function","branch":true,"nodes": [ ...
```

**Comment by phated** [comment]
Cherry-picked as 8e7064b807cf45de33a57775880596ab9fffa1b0 - thanks @sttk
